sdk/src/main/scala/com/horizen/api/http/SidechainCswApiRoute.scala

```diff
@@ -0,0 +1,163 @@
+import scala.util.{Failure, Success, Try}
+
+case class SidechainCswApiRoute(override val settings: RESTApiSettings,
+                                sidechainNodeViewHolderRef: ActorRef,
+                                cswManager: ActorRef)
+                               (implicit val context: ActorRefFactory, override val ec: ExecutionContext) extends SidechainApiRoute {
+
+  override val route: Route = pathPrefix("csw") {
+    hasCeased ~ generateCswProof ~ cswInfo ~ cswBoxIds ~ nullifier
+  }
+
+  /**
+   * Return ceasing status of the Sidechain
+   */
+  def hasCeased: Route = (post & path("hasCeased")) {
+    Try {
+      Await.result(cswManager ? GetCeasedStatus, timeout.duration).asInstanceOf[Boolean]
+    } match {
+      case Success(res) =>
+        ApiResponseUtil.toResponse(RespCswHasCeasedState(res))
+      case Failure(e) =>
+        log.error("Unable to retrieve ceasing status of the Sidechain.")
+        ApiResponseUtil.toResponse(ErrorRetrievingCeasingState("Unable to retrieve ceasing status of the Sidechain.", JOptional.of(e)))
+    }
+  }
+
+  /**
+   * Create a request for generation of CSW proof for specified
+   * and informs about current status of this proof
+   */
+  def generateCswProof: Route = (post & path("generateCswProof")) {
+    entity(as[ReqGenerationCswState]) { body =>
+      Try {
+        Await.result(cswManager ? GenerateCswProof(BytesUtils.fromHexString(body.boxId), body.mcAddress), timeout.duration).asInstanceOf[GenerateCswProofStatus]
+      } match {
+        case Success(res) =>
+          res match {
+            case SidechainIsAlive => ApiResponseUtil.toResponse(RespGenerationCswState("Sidechain is alive"))
+            case InvalidAddress  => ApiResponseUtil.toResponse(RespGenerationCswState("Invalid MC address"))
+            case NoProofData => ApiResponseUtil.toResponse(RespGenerationCswState("Sidechain is alive"))
+            case ProofGenerationStarted => ApiResponseUtil.toResponse(RespGenerationCswState("CSW proof generation is started"))
+            case ProofGenerationInProcess => ApiResponseUtil.toResponse(RespGenerationCswState("CSW proof generation in process"))
+            case ProofCreationFinished => ApiResponseUtil.toResponse(RespGenerationCswState("CSW proof generation is finished"))
+          }
+        case Failure(e) =>
+          log.error("Unexpected error during CSW proof generation.")
+          ApiResponseUtil.toResponse(ErrorCswGenerationState("Unexpected error during CSW proof generation.", JOptional.of(e)))
+      }
+    }
+  }
+
+  /**
+   * return the csw info for given BoxId
+   */
+  def cswInfo: Route = (post & path("cswInfo")) {
+     entity(as[ReqCswInfo]) { body =>
+       Try {
+         Await.result(cswManager ? GetCswInfo(BytesUtils.fromHexString(body.boxId)), timeout.duration).asInstanceOf[CswInfo]
+       } match {
+         case Success(cswInfo: CswInfo) =>
+           ApiResponseUtil.toResponse(RespCswInfo(cswInfo))
+         case Failure(e) =>
```

Comment:
We should separate Failure results from the CswManager from `Await` unexpected failures.
Same for `nullifier` endpoint.